### ISSUE TITLE
[telemetry] Instrument the UI page rendering path (AbstractApplicationXServletHandler + PhotonHTMLHelper)

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#41]
## Goal

Put a ph-telemetry `SERVER` span around the UI page rendering path, so that a full HTML request of a ph-oton application shows up as a trace next to the REST and AJAX entry points.

## Depends on

The `handleRequest` throws-clause widening issue — `AbstractApplicationXServletHandler.handleRequest` currently declares `throws IOException, ServletException`, which cannot host a `Telemetry.withSpanVoidThrowing` call. Do that one first.

## Where

**1. `ph-oton-core/src/main/java/com/helger/photon/core/servlet/AbstractApplicationXServletHandler.java`**

```java
public void handleRequest (@NonNull final IRequestWebScopeWithoutResponse aRequestScope,
                           @NonNull final UnifiedResponse aUnifiedResponse) throws ...
```

The body is just `createHTMLProvider (aRequestScope)` followed by `PhotonHTMLHelper.createHTMLResponse (...)`, wrapped in a `try` / `finally` that only does debug logging. Wrap it in the span.

**2. `ph-oton-app/src/main/java/com/helger/photon/app/html/PhotonHTMLHelper.java`**

```java
public static void createHTMLResponse (@NonNull final IRequestWebScopeWithoutResponse aRequestScope, ...)
```

This is the HC-tree → HTML serialization plus external CSS/JS resource merging (`mergeExternalCSSAndJSNodes`). It is also reachable from the AJAX path, so it deserves its own child span rather than being folded into (1).

## What to emit

New `com.helger.photon.core.CCoreTelemetry`:

| Constant | Value |
|---|---|
| `SPAN_PAGE_REQUEST` | `photon.page.request` |
| `METRIC_PAGE_REQUESTS` | `photon.page.requests` |
| `METRIC_PAGE_DURATION` | `photon.page.duration` |
| `ATTR_PAGE_SUCCESS` | `photon.page.success` |
| `UNIT_REQUEST` | `{request}` |
| `UNIT_MILLIS` | `ms` |

New `com.helger.photon.app.CAppTelemetry`:

| Constant | Value |
|---|---|
| `SPAN_HTML_RESPONSE` | `photon.html.response` |
| `METRIC_HTML_DURATION` | `photon.html.duration` |
| `ATTR_HTML_MIME_TYPE` | `photon.html.mimetype` |
| `UNIT_MILLIS` | `ms` |

Plus `CCoreMetrics` / `AppMetrics` instrument holders and package-private emitters, mirroring the `ph-oton-mgrs` `longrun` layout.

**Span (1)** `photon.page.request`, kind `SERVER` — the outer span for a UI request.
**Span (2)** `photon.html.response`, kind `INTERNAL` — a child of (1) when both are active, standalone when reached from AJAX.

## Cardinality

Do **not** put the request URL, query string, session ID or user ID on any metric attribute. The per-page dimension is handled separately by the `AbstractWebPage.getContent` issue, which has a properly bounded page ID; this issue's metrics stay global (no route dimension at all).

The MIME type from `PhotonHTMLHelper.getMimeType (aRequestScope)` is bounded (XHTML vs HTML) and is fine as a span attribute.

## POM

Both modules need a new dependency (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

`ph-oton-core` already gets `ph-telemetry` transitively via `ph-oton-mgrs`, but declare it directly since it is used directly. `ph-oton-app` has no transitive path and definitely needs it.

## Acceptance criteria

- [ ] Both spans emitted, nesting correctly when a UI request goes through both.
- [ ] Duration recorded for failed renders as well.
- [ ] Unit test with recording SPIs installed via `Telemetry.install (...)` / `TelemetryMetrics.install (...)` and reset to `null` in `@After`.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-core/.../core/CCoreTelemetry.java` | new, public - span / event / metric / attribute names |
| `ph-oton-core/.../core/servlet/PageRequestMetrics.java` | new, public - counter `photon.page.requests`, histogram `photon.page.duration` |
| `ph-oton-core/.../core/servlet/PageRequestTelemetry.java` | new, package-private emitter |
| `ph-oton-core/.../core/servlet/AbstractApplicationXServletHandler.java` | modified |
| `ph-oton-app/.../app/CAppTelemetry.java` | new, public |
| `ph-oton-app/.../app/html/HTMLResponseMetrics.java` | new, public - histogram `photon.html.duration` |
| `ph-oton-app/.../app/html/HTMLResponseTelemetry.java` | new, package-private emitter |
| `ph-oton-app/.../app/html/PhotonHTMLHelper.java` | modified |
| `ph-oton-core/pom.xml`, `ph-oton-app/pom.xml` | `com.helger.telemetry:ph-telemetry` added (no `<version>`, BOM) |
| `ph-oton-core/src/test/.../PageRequestTelemetryTest.java` | new, 3 tests - covers both modules and the nesting |

The instrument holder names are `PageRequestMetrics` / `HTMLResponseMetrics` rather than the `CCoreMetrics` / `AppMetrics` sketched in the issue: the `C` prefix is reserved for constant classes in this codebase, and naming them after the feature matches `LongRunningJobMetrics`.

### Two deliberate deviations from the issue text

**1. The spans are managed manually, not via `Telemetry.withSpanVoidThrowing`.**

`ForcedRedirectException` (Post-Redirect-Get) is a `RuntimeException`, and it is thrown straight through `createHTMLProvider (...)` / `createHTMLResponse (...)` on every successful form submission. All `Telemetry.withSpan*` helpers unconditionally `recordException (...)` + `setStatusError (...)` on any escaping `RuntimeException`, so using them would have marked **every successful form submit as a failed request** and skewed both the counter and the trace.

`XServletHandlerToSimpleHandler` already treats this exception specially (`catch (ForcedRedirectException ex) { /* Pass through */ throw ex; }`), and the instrumentation now does the same:

```java
try (final ITelemetrySpan aSpan = Telemetry.startSpan (CCoreTelemetry.SPAN_PAGE_REQUEST, ETelemetrySpanKind.SERVER))
{
  try { ... bSuccess = true; PageRequestTelemetry.onPageRequestSuccess (aSpan); }
  catch (final ForcedRedirectException ex) { bSuccess = true; PageRequestTelemetry.onPageRequestRedirect (aSpan); throw ex; }
  catch (final Exception ex)               { PageRequestTelemetry.onPageRequestError (aSpan, ex); throw ex; }
}
finally { PageRequestTelemetry.onPageRequestEnd (bSuccess, aSW.stopAndGetMillis ()); }
```

A redirect adds the span event `photon.page.forcedredirect`, sets `photon.page.success=true` and the span status to OK. New constant `CCoreTelemetry.EVENT_FORCED_REDIRECT`. Asserted by `testForcedRedirect`.

Side effect worth recording: this means the widening from #38 was not strictly *required* for this issue after all, because the wrapped body declares no checked exception. #38 still stands on its own - it restores the `IXServletSimpleHandler.handleRequest` contract that six classes had narrowed.

**2. `photon.html.response` covers the serialization only.**

`aHTMLProvider.createHTML (aRequestScope)` is where the `ForcedRedirectException` originates, and `ph-oton-app` does not have `ph-xservlet` on its classpath, so it cannot recognise that exception type. Rather than adding a dependency to a lower layer, the HC tree creation was moved **before** the span is started. The span now measures HC tree → bytes → `UnifiedResponse` only.

That is arguably the better decomposition anyway: the tree creation time is already covered by the outer `photon.page.request` span and, per screen, by `photon.webpage.duration` from #42. `CAppTelemetry.SPAN_HTML_RESPONSE` and the histogram description document the narrowed scope.

### Emission

* Span `photon.page.request` (`SERVER`), attribute `photon.page.success`, event `photon.page.forcedredirect`.
* Counter `photon.page.requests` + histogram `photon.page.duration`, both with `photon.page.success` only - **no route dimension**, as required.
* Span `photon.html.response` (`INTERNAL`), attributes `photon.html.mimetype` (`getAsStringWithoutParameters ()`) and `photon.html.success`.
* Histogram `photon.html.duration` with `photon.html.success`. `photon.html.success` was added beyond the constants listed in the issue so that failed serializations can be separated in the histogram.

### Verification

`PageRequestTelemetryTest` runs a real `AbstractApplicationXServletHandler` and asserts:

* successful request → exactly **2 spans**, `photon.page.request` (`SERVER`) started first and `photon.html.response` (`INTERNAL`) nested inside it, both closed and OK, `photon.page.requests` metric with exactly one attribute;
* failing request → only the outer span, status `error`, exception recorded, counter and histogram both carrying `photon.page.success=false`;
* forced redirect → span status `ok`, no exception recorded, event present, counter carrying `photon.page.success=true`.

Full reactor `mvn install` on all 23 modules: BUILD SUCCESS, no test failures. `README.md` news bullets added under the new `v10.5.1 - work in progress` section.
